### ISSUE TITLE
Plugin `check_vmware_datastore_performance` incorrectly indicates `UNKNOWN` state

### ISSUE THREAD

**Issue by atc0005** [atc0005/check-vmware#542]
Partial output from a service check I setup earlier this AM (and let run throughout the day):

```log
UNKNOWN: Datastore HUSVM-DC1-vol6 (6 VMs) exceeds specified performance thresholds: [WriteLatency]

**ERRORS**

* datastore latency metric sets are all value zero

**THRESHOLDS**

* CRITICAL: { Percentile: 90, ReadLatency: 30, WriteLatency: 30, VMLatency: 30 }
* WARNING: { Percentile: 90, ReadLatency: 15, WriteLatency: 15, VMLatency: 15 }

**DETAILED INFO**

Metrics for datastore "HUSVM-DC1-vol6" which exceed thresholds:

Result 1 (active):
* { Percentile: 90, RLatency: 4.49, WLatency: 21.13, VMLatency: 0.23, RIops: 7.00, WIops: 0.00, Interval: 83540 }

Full Collection of Performance Metrics for datastore "HUSVM-DC1-vol6" (6 VMs):

Result 1 (active):
* { Percentile: 50, RLatency: 0.10, WLatency: 0.47, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 83540 }
* { Percentile: 60, RLatency: 0.19, WLatency: 0.91, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 83540 }
* { Percentile: 70, RLatency: 0.40, WLatency: 1.86, VMLatency: 0.05, RIops: 1.00, WIops: 0.00, Interval: 83540 }
* { Percentile: 80, RLatency: 0.05, WLatency: 4.74, VMLatency: 0.11, RIops: 3.00, WIops: 0.00, Interval: 83540 }
* { Percentile: 90, RLatency: 4.49, WLatency: 21.13, VMLatency: 0.23, RIops: 7.00, WIops: 0.00, Interval: 83540 }

Result 2 (historical):
* { Percentile: 50, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85320 }
* { Percentile: 60, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85320 }
* { Percentile: 70, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85320 }
* { Percentile: 80, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85320 }
* { Percentile: 90, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85320 }

Result 3 (historical):
* { Percentile: 50, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85180 }
* { Percentile: 60, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85180 }
* { Percentile: 70, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85180 }
* { Percentile: 80, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85180 }
* { Percentile: 90, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85180 }

Result 4 (historical):
* { Percentile: 50, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85340 }
* { Percentile: 60, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85340 }
* { Percentile: 70, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85340 }
* { Percentile: 80, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85340 }
* { Percentile: 90, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85340 }

Result 5 (historical):
* { Percentile: 50, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85320 }
* { Percentile: 60, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85320 }
* { Percentile: 70, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85320 }
* { Percentile: 80, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85320 }
* { Percentile: 90, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85320 }

Result 6 (historical):
* { Percentile: 50, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85240 }
* { Percentile: 60, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85240 }
* { Percentile: 70, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85240 }
* { Percentile: 80, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85240 }
* { Percentile: 90, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85240 }

Result 7 (historical):
* { Percentile: 50, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85300 }
* { Percentile: 60, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85300 }
* { Percentile: 70, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85300 }
* { Percentile: 80, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85300 }
* { Percentile: 90, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85300 }

Result 8 (historical):
* { Percentile: 50, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85400 }
* { Percentile: 60, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85400 }
* { Percentile: 70, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85400 }
* { Percentile: 80, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85400 }
* { Percentile: 90, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85400 }
```

Of note:

- UNKNOWN state triggered instead of `WARNING` or `CRITICAL` due to crossing write latency threshold
- historical values *are* all zero
- active values are *not* all zero

**Comment by atc0005** [comment]
After a bit, the state remained `UNKNOWN`, but the one-line summary changed:

```log
UNKNOWN: Datastore HUSVM-DC1-vol6 (6 VMs) performance metrics (for all intervals) are unavailable

**ERRORS**

* datastore latency metric sets are all value zero

**THRESHOLDS**

* CRITICAL: { Percentile: 90, ReadLatency: 30, WriteLatency: 30, VMLatency: 30 }
* WARNING: { Percentile: 90, ReadLatency: 15, WriteLatency: 15, VMLatency: 15 }

**DETAILED INFO**

Full Collection of Performance Metrics for datastore "HUSVM-DC1-vol6" (6 VMs):

Result 1 (active):
* { Percentile: 50, RLatency: 0.10, WLatency: 0.46, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 84440 }
* { Percentile: 60, RLatency: 0.01, WLatency: 0.03, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 84440 }
* { Percentile: 70, RLatency: 0.40, WLatency: 1.82, VMLatency: 0.06, RIops: 1.00, WIops: 0.00, Interval: 84440 }
* { Percentile: 80, RLatency: 1.02, WLatency: 4.64, VMLatency: 0.13, RIops: 3.00, WIops: 0.00, Interval: 84440 }
* { Percentile: 90, RLatency: 4.40, WLatency: 0.07, VMLatency: 0.25, RIops: 7.00, WIops: 0.00, Interval: 84440 }

Result 2 (historical):
* { Percentile: 50, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85320 }
* { Percentile: 60, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85320 }
* { Percentile: 70, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85320 }
* { Percentile: 80, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85320 }
* { Percentile: 90, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85320 }

Result 3 (historical):
* { Percentile: 50, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85180 }
* { Percentile: 60, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85180 }
* { Percentile: 70, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85180 }
* { Percentile: 80, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85180 }
* { Percentile: 90, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85180 }

Result 4 (historical):
* { Percentile: 50, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85340 }
* { Percentile: 60, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85340 }
* { Percentile: 70, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85340 }
* { Percentile: 80, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85340 }
* { Percentile: 90, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85340 }

Result 5 (historical):
* { Percentile: 50, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85320 }
* { Percentile: 60, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85320 }
* { Percentile: 70, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85320 }
* { Percentile: 80, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85320 }
* { Percentile: 90, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85320 }

Result 6 (historical):
* { Percentile: 50, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85240 }
* { Percentile: 60, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85240 }
* { Percentile: 70, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85240 }
* { Percentile: 80, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85240 }
* { Percentile: 90, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85240 }

Result 7 (historical):
* { Percentile: 50, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85300 }
* { Percentile: 60, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85300 }
* { Percentile: 70, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85300 }
* { Percentile: 80, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85300 }
* { Percentile: 90, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85300 }

Result 8 (historical):
* { Percentile: 50, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85400 }
* { Percentile: 60, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85400 }
* { Percentile: 70, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85400 }
* { Percentile: 80, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85400 }
* { Percentile: 90, RLatency: 0.00, WLatency: 0.00, VMLatency: 0.00, RIops: 0.00, WIops: 0.00, Interval: 85400 } 
```

The UNKNOWN state is invalid here. Based on these details, the plugin should be declaring an `OK` state as the thresholds are not being crossed.

**Comment by atc0005** [comment]
I've identified what I believe is a fix for both issues. 

I plan to apply the fix tomorrow AM and then release `v0.27.0-rc2` with the changes.
